### PR TITLE
annotations-runtime conversion from Kotlin to Java

### PR DESCRIPTION
**Pull request by kerkhofsd**

### CODE REVIEW

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+import org.springframework.extensions.webscripts.TransactionParameters;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * Creates {@link AnnotationWebScript} instances from beans defined in a {@link BeanFactory}.
+ *
+ * @author Laurens Fridael
```
Comment (todorinskiz) [inline annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java]: Should we remove the author annotations as discussed in other projects?
They tend to get out of date really fast, and give wrong impression of the maintenance ownership.

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+ * @author Laurent Van der Linden
+ */
+public final class AnnotationWebScriptBuilder implements BeanFactoryAware {
+
+    @Autowired
+    private HandlerMethodArgumentsResolver handlerMethodArgumentsResolver;
+    @Autowired
+    private MessageConverterRegistry messageConverterRegistry;
+
+    public final void setHandlerMethodArgumentsResolver(@NotNull HandlerMethodArgumentsResolver value) {
```
Comment (todorinskiz) [inline annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java]: @NotNull is not enough? 😮

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+                            throw new RuntimeException(
+                                    "Cannot combine @Before, @Attribute and @Uri on a single method. Method: "
+                                            + ClassUtils.getQualifiedMethodName(method));
+                        }
+                        handlerMethods.getBeforeMethods().add(method);
+                    }
+                }
+        );
+        ReflectionUtils.doWithMethods(beanType, (method) -> {
+                    Intrinsics.checkParameterIsNotNull(method, "method");
```
Comment (todorinskiz) [inline annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java]: Please refactor this anonymous method to a named one ... 😿

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+                        }
+                        if (Void.TYPE.equals(method.getReturnType())) {
+                            throw new RuntimeException("@Attribute methods cannot have a void return type.");
+                        }
+                        handlerMethods.getAttributeMethods().add(method);
+                    }
+                }
+        );
+        ReflectionUtils.doWithMethods(beanType, (method) -> {
+                    Intrinsics.checkParameterIsNotNull(method, "method");
```
Comment (todorinskiz) [inline annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java]: Please refactor this anonymous method to a named one ... 😿

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+                                            + ClassUtils.getQualifiedMethodName(method));
+                        }
+                        handlerMethods.getExceptionHandlerMethods().add(new ExceptionHandlerMethod(exceptionHandler, method));
+                    }
+                }
+        );
+
+        final ArrayList<org.springframework.extensions.webscripts.WebScript> webScripts = new ArrayList<>();
+        ReflectionUtils.doWithMethods(beanType, (method) -> {
+                    Intrinsics.checkParameterIsNotNull(method, "method");
```
Comment (todorinskiz) [inline annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java]: Please refactor this anonymous method to a named one ... 😿

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java (hunk 2 above)
Comment (tgeens) [reply annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptBuilder.java]: @todorinskiz the annotation doesn't change the runtime behavour

**Pull request merged**